index.js

```diff
@@ -137,21 +150,68 @@ function useCard()
     //Play card if valid move, otherwise ignore
     if (isValidCard == true)
     {
+        let cardBeingPlayed = players[gameTurn].playerDeck.getCard(cardIndex);
         alert("Debug: Valid move.");
+        
+        
+        //Will run if there is a stackable card played, +2 or +4
+        if(drawStack.stackAmt != 0){
+                if(cardBeingPlayed.value != drawStack.cardValue){
+                    alert("Card chosen Doesn't stack");
+                    return;
+                }else if(cardBeingPlayed.value == 1 && cardBeingPlayed.color != 'Special'){
+                    alert("Card chosen Doesn't stack");
+                    return;
+                }
+                
+        }
+        
         players[gameTurn].playerDeck.playCard(cardIndex);
-        gameTurn++;
+        
+        if(cardBeingPlayed.color == 'Special'){
+            if(cardBeingPlayed.value == 0){
+                cardWild();
+            }else if(cardBeingPlayed.value == 1){
+                cardDraw4();
+            }
+        }else if(cardBeingPlayed.value == 10){
+            cardDraw2();
+        }else if(cardBeingPlayed.value == 11){
+            cardReverse();
```

Comment:
This is hard to test with only 2 players, but didn't notice anything wrong with the code. We should probably test this again when we have 3 players.

Reply:
will test when we expand to 3-4 players